sycl/test-e2e/lit.cfg.py

```diff
@@ -799,12 +761,51 @@ def open_check_file(file_name):
     # Use short names for LIT rules.
     features.add(be)
 
+    if be == "hip" and config.hip_platform == "AMD":
+        if not config.amd_arch:
+            arch = ""
+            for a in architecture_feature:
+                arch = a
+            amd_arch_prefix = "arch-amd_gpu_"
+            if amd_arch_prefix not in arch or len(architecture_feature) != 1:
+                lit_config.error(
+                    "Cannot detect architecture for AMD HIP device, specify it explicitly"
+                )
+            config.amd_arch = arch.replace(amd_arch_prefix, "")
```

Comment:
Is there a bug with indentation here? Otherwise I don't see a reason behind the loop at line 767...

Reply:
It's because `architecture_feature` is a set and there's no easy way to get the first element out I think, indexing doesn't work.  Another option is next(iter(architecture_feature)) I think.

But I'm not good at python so maybe there's something I missed

Reply:
```
# Guaranteed to be a sinle element in the set
arch = [x for x in architecture_feature][0]
```

Or better yet not use set to start with :)
```

Reply:
thanks, ill fix it.

the existing code is using a set, probably for some reason, and i dont want to touch it :P

Reply:
Yeah, I think it shouldn't have been a set at all. Someone probably just copied aspects/sg_sizes but that wasn't a good idea.

Reply:
tale as old as time